Framework/Kernel/test/PropertyManagerPropertyTest.h

```diff
@@ -0,0 +1,163 @@
+    TS_ASSERT_DIFFERS(retrieved, emptyMgr);
+  }
+
+  void test_Property_Can_Implicitly_Convert_To_Value_Type() {
+    using Mantid::Kernel::PropertyManager;
+    using Mantid::Kernel::PropertyManager_sptr;
+    // Create a top-level PropertyManager, add a PropertyManagerProperty
+    // as a nested PropertyManager
+    auto topMgr = boost::make_shared<PropertyManager>();
+    topMgr->declareProperty(
+        Mantid::Kernel::make_unique<PropertyManagerProperty>("Args"));
+    topMgr->setProperty("Args", createPropMgrWithInt());
+
+    PropertyManager_sptr args = topMgr->getProperty("Args");
+  }
+
+  void test_Property_Set_With_Json_String_Overwrites_Existing_Values() {
+    PropertyManagerProperty prop("Test", createPropMgrWithInt());
+
+    auto secondMgr = createPropMgrWithInt();
+    secondMgr->setProperty("Prop1", 5);
+    TS_ASSERT_EQUALS("", prop.setValue(secondMgr->asString(true)));
+    TS_ASSERT_EQUALS(secondMgr->asString(true), prop.value());
+    auto retrieved = prop();
+    TS_ASSERT_EQUALS(1, retrieved->propertyCount());
+    TS_ASSERT(prop()->existsProperty("Prop1"));
+  }
+
+  void test_Property_Set_With_String_Checks_PropertyManager_DataService() {
+    using Mantid::Kernel::PropertyManagerDataService;
+    auto globalMgr = createPropMgrWithInt();
+    const std::string pmdsName("globalprops");
+    auto &pmds = PropertyManagerDataService::Instance();
+    pmds.add(pmdsName, globalMgr);
+    PropertyManagerProperty prop("PMDSTest");
+
+    TS_ASSERT_EQUALS("", prop.setValue(pmdsName));
+    TS_ASSERT_EQUALS(pmdsName, prop.value());
+    auto retrieved = prop();
+    TS_ASSERT_EQUALS(1, retrieved->propertyCount());
+    TS_ASSERT(prop()->existsProperty("Prop1"));
+
+    pmds.remove(pmdsName);
+  }
+
+  void test_Property_Set_As_DataService_Key_Then_Json_Returns_Correct_String() {
+    using Mantid::Kernel::PropertyManagerDataService;
+    auto globalMgr = createPropMgrWithInt();
+    const std::string pmdsName("globalprops");
+    auto &pmds = PropertyManagerDataService::Instance();
+    pmds.add(pmdsName, globalMgr);
+
+    PropertyManagerProperty prop("PMDSTest");
+    prop.setValue(pmdsName);
+    TS_ASSERT_EQUALS(pmdsName, prop.value());
+    pmds.remove(pmdsName);
+
+    TS_ASSERT_EQUALS("", prop.setValue(globalMgr->asString(true)));
+    TS_ASSERT_EQUALS(globalMgr->asString(true), prop.value());
+  }
+
+  //----------------------------------------------------------------------------
+  // Failure tests
+  //----------------------------------------------------------------------------
+  void test_Empty_Name_Is_Not_Accepted() {
+    TS_ASSERT_THROWS(PropertyManagerProperty(""), std::invalid_argument);
+  }
+
+  void test_Empty_Property_Set_With_Json_String_Returns_Error() {
+    PropertyManagerProperty prop("Test");
+
+    auto json = createPropMgrWithInt()->asString(true);
+    std::string msg;
+    TS_ASSERT_THROWS_NOTHING(msg = prop.setValue(json));
+    TS_ASSERT(msg.length() > 0);
+  }
```

Comment:
This is unfortunate, I was hoping this would be simpler to achieve but after I thought a little the implementation was going to be quite convoluted for little gain.

The problem with having an empty manager and it only being passed a string is that I have no idea what types all of the individual elements should be so it would have to be inferred which is difficult and error prone.

I see this being most useful from Python when translating an existing `dict` and in this case we will then know the types of the elements and can translate this properly. This is done in #16029.